thavalon-server/src/game/mod.rs

```diff
@@ -169,6 +169,11 @@ impl Game {
     pub fn size(&self) -> usize {
         self.proposal_order.len()
     }
+
+    /// Look up the display name associated with a given role, if it exists.
+    pub fn display_name_from_role(&self, role: Role) -> Option<&String> {
+        self.info.keys().find(|&x| self.info.get(x).unwrap().get_role() == role)
```

Comment:
We should use `.iter()` to avoid looking up keys in the map while iterating over it:
```
self.info.iter().find_map(|(&player, info)| if info.get_role() == role { Some(player) } else { None })
```